src/storage-controller/src/lib.rs

```diff
@@ -1643,26 +1668,68 @@ where
                 }
             }
 
-            // Check if the collection is for a Webhook source, unregister if so.
+            // Check if a collection is managed by the storage-controller itself.
             let collection = self.collections.get(&id);
             if let Some(CollectionState { description, .. }) = collection {
-                if description.data_source == DataSource::Webhook && frontier.is_empty() {
-                    // Unregister our collection from the manager so writes should no longer occur.
-                    self.collection_manager.unregister_collection(id).await;
+                let is_controller_managed = matches!(
```

Comment:
I basically always do a full `match` on enums so that adding a variant generates a compile error and the code author is directed to all the various places they need to consider.

Also, should this be a method on `DataSource`?

Reply:
Good call, I refactored this a bit, let me know what you think!